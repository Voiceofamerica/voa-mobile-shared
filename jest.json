{
  "setupTestFrameworkScriptFile": "<rootDir>/.jest/setupTests.js",
  "transform": {
    "\\.(ts|tsx)$": "ts-jest/preprocessor.js",
    "\\.(js|jsx)$": "babel-jest"
  },
  "setupFiles": [
    "<rootDir>/.jest/rAF.js"
  ],
  "testRegex": "\\.spec\\.(ts|tsx)$",
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js",
    "json"
  ],
  "cacheDirectory": "jestCache",
  "globals": {
    "__DEV__": true,
    "ts-jest": {
      "tsConfigFile": "tsconfig.test.json"
    }
  },
  "moduleNameMapper": {
    "^components/(.*)$": "<rootDir>/components/$1/index.ts",
    "^helpers/(.*)$": "<rootDir>/helpers/$1.ts",
    "^types/(.*)$": "<rootDir>/styles/$1.d.ts",
    "\\.s?css$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|svg)$": "<rootDir>/__mocks__/fileMock.js"
  },
  "collectCoverage": true,
  "collectCoverageFrom": [
    "**/*.{ts,tsx}",
    "!**/index.{ts,tsx}",
    "!**/*.story.tsx",
    "!**/*.d.ts"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 45,
      "functions": 40,
      "lines": 50,
      "statements": 50
    }
  }
}