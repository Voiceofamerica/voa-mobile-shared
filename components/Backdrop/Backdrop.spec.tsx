
import 'jest'
import 'react-native'
import * as React from 'react'
import * as RX from 'reactxp'
import * as renderer from 'react-test-renderer'
import { shallow, configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
configure({ adapter: new Adapter() })

import OriginalBackdrop from '.'

const importModule = (): typeof OriginalBackdrop => {
  const module = require('.').default
  return module
}

const mockedComponents = [
]

const defaultMocks = [
]

const allMocks = mockedComponents.concat(defaultMocks)

describe('<Backdrop />', () => {
  beforeEach(() => {
    jest.resetModules()
    mockedComponents.forEach(component => {
      jest.setMock(component, component)
    })
    defaultMocks.forEach(component => {
      jest.setMock(component, jest.genMockFromModule(component))
    })
  })

  afterEach(() => {
    allMocks.forEach(component => {
      jest.unmock(component)
    })
  })

  describe('snapshots', () => {
    it('renders correctly without blur', () => {
      const Backdrop = importModule()
      const tree = renderer.create(
        <Backdrop imgSrc={1} />,
      )

      expect(tree).toMatchSnapshot()
    })

    it('renders correctly with blur', () => {
      const Backdrop = importModule()
      const tree = renderer.create(
        <Backdrop imgSrc={1} blur />,
      )

      expect(tree).toMatchSnapshot()
    })
  })
})
