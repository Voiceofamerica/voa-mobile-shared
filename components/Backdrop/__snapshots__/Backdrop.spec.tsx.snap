// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Backdrop /> snapshots renders correctly with blur 1`] = `
<RX.View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <Image
    defaultSource={undefined}
    onLoadEnd={[Function]}
    resizeMode="cover"
    source={1}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</RX.View>
`;

exports[`<Backdrop /> snapshots renders correctly without blur 1`] = `
<RX.View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <Image
    defaultSource={undefined}
    onLoadEnd={[Function]}
    resizeMode="cover"
    source={1}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  />
</RX.View>
`;
